src/averageAges.js

```diff
@@ -1,61 +1,84 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
+
+function calculateAge(person) {
+  return person.died - person.born;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredAges = people.filter((person) =>
+    person.sex === 'm'
+    && (century ? Math.ceil(person.died / 100) === century : true)
+  );
+
+  return filteredAges.reduce((sum, person) =>
+    sum + calculateAge(person), 0) / filteredAges.length;
 }
 
-/**
+/*
  * Implement calculateWomenAverageAge function
  *
  * Function returns average age of women in array. If `withChildren` is
  * specified then function calculates average age only for women with children
  *
  * Hint: To check if a woman has children you should find someone who mention
  * her as mother.
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const findWomen = people.filter(item => item.sex === 'f');
+  const motherNames = people.map(item => item.mother);
+
+  const selectedWomen = findWomen.filter(item =>
+    motherNames.includes(item.name));
+
+  const sumOfAges = (withChildren === true)
+    ? selectedWomen.map(item => item.died - item.born)
+    : findWomen.map(item => item.died - item.born);
+
+  return sumOfAges.reduce((a, b) => (a + b), 0) / sumOfAges.length;
 }
 
 /**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
  * @param {object[]} people
  * @param {boolean} onlyWithSon - optional
  *
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const findMother = people.filter(item => {
+    return people.find(person => person.mother === item.name);
+  });
+
+  const relatives = onlyWithSon
+    ? people.filter(child => {
```

Comment:
Consider renaming 'item' to 'person' for better readability and consistency with the rest of the code.